meilisearch-http/src/helpers/compression.rs

```diff
@@ -1,19 +1,27 @@
 use flate2::Compression;
 use flate2::read::GzDecoder;
 use flate2::write::GzEncoder;
-use std::fs::{create_dir_all, File};
+use std::fs::{create_dir_all, rename, File};
 use std::path::Path;
 use tar::{Builder, Archive};
+use uuid::Uuid;
 
 use crate::error::Error;
 
 pub fn to_tar_gz(src: &Path, dest: &Path) -> Result<(), Error> {
-    let f = File::create(dest)?;
+    let file_name = format!(".{}", Uuid::new_v4().to_urn());
```

Comment:
I think adding Uuid as a dependency for this purpose is a bit overkill, a timestamp should be largely enough, and chrono is already a dependency
```suggestion
    let file_name = format!(".{}", Utc::now().timestamp_nanos());
```

Reply:
I believe that is a valid concern that this introduces a new dependency (although `uuid` is already a dependency for the overall project) and of course it can be changed to the `chrono` usage.

The reason why I went with the `uuid` solution is security. The UUIDv4s cannot be guessed and an attack vector could be eliminated with this solution. If we change the implementation to the timestamp way then an attacker could create a file (or rather multiple files) and get ownership of the final snapshot file after the rename and overwrite with arbitrary content.

If you feel this is not a real concern since the snapshot directory should be protected by correct ownership then I will change the code to your suggestion of course.

Reply:
Well if an attacker can create an arbitrary file on your filesystem, he can also create one that is named like your database don't you think? 🙂 Imo even "temp" would be enough for this file name, but since uuid is already a dep, I don't have any strong opinion

Reply:
If the snapshot file does not exist yet on the system then you are absolutely right. So what should I do? Accept the change or is it good as it is now? 😄

Reply:
@LegendreM what do you think?

Reply:
Hello @raszi, thanks for your PR, I will try it locally before, but I think we will merge the current state 👍

Reply:
Any update on this?